src/test/burn/TestData/FilesInUseTests/BundleA/BundleA.wixproj

```diff
@@ -3,9 +3,10 @@
   <PropertyGroup>
     <OutputType>Bundle</OutputType>
     <UpgradeCode>{6A348108-8ACE-4D13-A352-D8F76785BFE4}</UpgradeCode>
+    <DefineConstants>$(DefineConstants);BaseOutputPath=$(BaseOutputPath);Version=1.1</DefineConstants>
   </PropertyGroup>
   <ItemGroup>
-    <Compile Include="..\..\Templates\Bundle.wxs" Link="Bundle.wxs" />
+    <Compile Include="..\..\WixStdBaTests\BundleA\Bundle.wxs" Link="Bundle.wxs" />
```

Comment:
I don't have time for a full review right now, but since you seem to be trying to get the E2E test working:

You can't swap out TestBA for wixstdba when the test contains TestBAController. TestBAController writes things to the registry that modifies the behavior of TestBA. You have to create your own bundle and your own test unless you want to create a BAFunctions.dll that implements the TestBAController directives.

Reply:
Yes, I finally figured that out. Actually, it works, mostly. :) I put the original test back and added one specifically for WixStdBA. Testing now.